packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialog.java

```diff
@@ -558,8 +563,15 @@ private PowerAction() {
         public boolean onLongPress() {
             UserManager um = (UserManager) mContext.getSystemService(Context.USER_SERVICE);
             if (!um.hasUserRestriction(UserManager.DISALLOW_SAFE_BOOT)) {
-                mWindowManagerFuncs.reboot(true, null);
-                return true;
+                if (mKeyguardMonitor.isSecure() && mKeyguardMonitor.isShowing() && mSecure){
```

Comment:
code looking good, but having a feeling this could have been turned into a function
seems too repetitive

Reply:
Yeah, I didn't think about doing a method before, will do